microprofile/microprofile-telemetry/src/main/java/org/wildfly/extension/microprofile/telemetry/MicroProfileTelemetryDeploymentProcessor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.microprofile.telemetry;
+
+import static org.jboss.as.weld.Capabilities.WELD_CAPABILITY_NAME;
+import static org.wildfly.extension.microprofile.telemetry.MicroProfileTelemetryExtensionLogger.MPTEL_LOGGER;
+
+import java.util.function.Supplier;
+
+import io.smallrye.opentelemetry.api.OpenTelemetryConfig;
+import org.jboss.as.controller.capability.CapabilityServiceSupport;
+import org.jboss.as.ee.structure.DeploymentType;
+import org.jboss.as.ee.structure.DeploymentTypeMarker;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.weld.WeldCapability;
+import org.wildfly.extension.microprofile.telemetry.api.MicroProfileTelemetryCdiExtension;
+
+public class MicroProfileTelemetryDeploymentProcessor implements DeploymentUnitProcessor {
+    @Override
+    public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
+        MPTEL_LOGGER.activatingSubsystem();
+
+        final DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
+        if (DeploymentTypeMarker.isType(DeploymentType.EAR, deploymentUnit)) {
+            return;
+        }
+
+        try {
+            final CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
```

Comment:
Is ```support``` guaranteed to be not null? If not, a [Assert.checkNotNullParam](https://github.com/wildfly/wildfly-common/blob/main/src/main/java/org/wildfly/common/Assert.java#L47) makes the Exception better readable and explicit.

Reply:
Fixed